src/Core/VersionChecker.cs

```diff
@@ -1,15 +1,38 @@
 using System.Net.Http.Json;
-using System.Reflection;
 using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Product;
 
 namespace Azure.DataApiBuilder.Core;
 
+/// <summary>
+/// A class to test the local version of the engine against the NuGet version.
+/// </summary>
+/// <remarks>
+/// This is used in startup to suggest upgrading the engine.
+/// </remarks>
 public static class VersionChecker
 {
-    public static void GetVersions(out string? latestVersion, out string? currentVersion)
+    private const string NUGETURL = "https://api.nuget.org/v3-flatcontainer/microsoft.dataapibuilder/index.json";
+
+    /// <summary>
+    /// Checks if the current local version of the product matches the latest version available on NuGet.
+    /// </summary>
+    /// <param name="nugetVersion">Outputs the latest version available on NuGet.</param>
+    /// <param name="localVersion">Outputs the current local version of the product.</param>
+    /// <returns>
+    /// Returns <c>true</c> if the local version matches the latest NuGet version or if the NuGet version is not available;
+    /// otherwise, returns <c>false</c>.
+    /// </returns>
+    /// <remarks>
+    //  If the internet is unavailable or NuGet is down or the HTTP request fails for any reason
+    /// (there is a 2 second Timeout on the request), then the NuGet version will be <c>null</c> and
+    //  this method will return <c>true</c>. This is mostly because this check is a user courtesy.
+    // </remarks>
+    public static bool IsCurrentVersion(out string? nugetVersion, out string? localVersion)
     {
-        latestVersion = FetchLatestNuGetVersion();
-        currentVersion = GetCurrentVersionFromAssembly(Assembly.GetCallingAssembly());
+        nugetVersion = FetchLatestNuGetVersion();
+        localVersion = ProductInfo.GetProductVersion(false);
+        return string.IsNullOrEmpty(nugetVersion) || nugetVersion == localVersion;
```

Comment:
What if the localVersion is > than the nugetVersion? Could happen when people clone our repo and we are in the process of developing 1.5, But nuget version will be 1.4 in that timeframe.